python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,373 @@
+            properties=properties,
+        )
+        serialized_json = request.json()
+        response = requests.post(
+            self.url(Endpoints.create_table, namespace=namespace_and_table["namespace"]),
+            data=serialized_json,
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {409: TableAlreadyExistsError})
+
+        return RestTable(identifier=identifier, **response.json())
+
+    def list_tables(self, namespace: Optional[Union[str, Identifier]] = None) -> List[Identifier]:
+        namespace_concat = NAMESPACE_SEPARATOR.join(self.identifier_to_tuple(namespace or ""))
+        response = requests.get(
+            self.url(Endpoints.list_tables, namespace=namespace_concat),
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError})
+        response = ListTablesResponse(**response.json())
+
+        return [(*entry.namespace, entry.name) for entry in response.identifiers]
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        response = requests.get(
+            self.url(Endpoints.load_table, prefixed=True, **self._split_namespace_and_table(identifier)), headers=self.headers
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError})
+        return Table(identifier=identifier, **response.json())
+
+    def drop_table(self, identifier: Union[str, Identifier], purge_requested: bool = False) -> None:
+        response = requests.delete(
+            self.url(Endpoints.drop_table, prefixed=True, purge=purge_requested, **self._split_namespace_and_table(identifier)),
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError})
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        raise NotImplementedError("Not implemented")
+
+    def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: Union[str, Identifier]):
+        payload = {
+            "source": self._split_namespace_and_name(from_identifier),
+            "destination": self._split_namespace_and_name(to_identifier),
+        }
+        response = requests.post(self.url(Endpoints.rename_table), json=payload, headers=self.headers)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {404: NoSuchTableError, 409: TableAlreadyExistsError})
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Optional[Properties] = None) -> None:
```

Comment:
Should this send `{}` instead of `None`? I'm not sure which one is better.

Reply:
This is because the properties aren't required by the spec: https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml

I think `{}` is better than `None`.

Reply:
I've updated the code: https://github.com/apache/iceberg/pull/5358

Reply:
I've fixed this in the line below:
```python
def create_namespace(self, namespace: Union[str, Identifier], properties: Optional[Properties] = None) -> None:
    payload = {"namespace": self.identifier_to_tuple(namespace), "properties": properties or {}}
```